honeybee_3dm/config.py

```diff
@@ -0,0 +1,432 @@
+    """
+
+    if 'grid_settings' in config['layers'][layer_name] and \
+            config['layers'][layer_name]['exclude_from_rad']:
+        grid_controls = config['layers'][layer_name]['grid_settings']
+        return grid_controls['grid_size'], grid_controls['grid_offset']
+    else:
+        return None
+
+
+def check_parent_in_config(file_3dm, config, layer_name, parent_layer_name):
+    """Checks if the parent layer of a layer is already mentioned in the config file.
+
+    This function will return True if the parent layer of a layer is already mentioned
+    in the config file and child layers are requested from that parent layer in the
+    config file.
+
+    Args:
+        file_3dm: A rhino3dm file objects.
+        config: A dictionary of the config settings.
+        layer_name: A text string of the layer name
+        parent_layer_name: A text string of the parent layer name
+
+    Returns:
+        A bool.
+    """
+
+    if parent_layer_name in config['layers'] and\
+        'include_child_layers' in config['layers'][parent_layer_name] and\
+        config['layers'][parent_layer_name]['include_child_layers'] == True:
+        return True
+    else:
+        return False
+
+
+def face3d_to_hb_face_with_face_type(config, face_obj, name, layer_name):
+    """Create a Honeybee Face object with a specific face_type.
+
+    This function returns a Honeybee Face object with a specific face_type requested
+    in the config file and also assign a radiance material to the face if requested
+    from the config file.
+
+    Args:
+        config: A dictionary of the config settings.
+        face_obj: A Ladybug Face3d object.
+        name: A text string of the name of the rhino object.
+        layer_name: A text string of the rhino layer name.
+
+    Returns:
+        A Honeybee Face object.
+    """
+    
+    obj_name = name or clean_and_id_string(layer_name)
+    args = [clean_string(obj_name), face_obj]
+    
+    if config['layers'][layer_name]['honeybee_face_type'] == 'roof':
+        face_type = face_types.roof_ceiling
+    
+    elif config['layers'][layer_name]['honeybee_face_type'] == 'wall':
+        face_type = face_types.wall
+    
+    elif config['layers'][layer_name]['honeybee_face_type'] == 'floor':
+        face_type = face_types.floor
+    
+    elif config['layers'][layer_name]['honeybee_face_type'] == 'airwall':
+        face_type = face_types.air_boundary
+    
+    args.append(face_type)
+    hb_face = Face(*args)
+    hb_face.display_name = args[0]
+    
+    if 'radiance_material' in config['layers'][layer_name]:
+        radiance_modifiers = mat_to_dict(config['sources']['radiance_material'])
+        hb_face.properties.radiance.modifier = radiance_modifiers[config
+            ['layers'][layer_name]['radiance_material']]
+        return hb_face
+    else:
+        return hb_face
+
+
+def face3d_to_hb_face_with_rad(config, face_obj, name, layer_name):
```

Comment:
You should probably move all these methods to a different module and keep this module limited to Config.

Reply:
Moved